### PR TITLE
feat(RHINENG-3070): implement two tabs in pathway details page

### PR DESCRIPTION
**Pull request by mkholjuraev**
# Description

Associated Jira ticket: # RHINENG-3070

This creates hybrid tabs for the pathways details page.

Acceptance criteria:

**Important: Immutable tab should behave as it is in recommendation detail page without table row actions**

1. Select the system number from a Pathway row on the main page and go to the Pathway details. If there are any conventional systems, land on details with Conventional systems in view ([Table](https://www.sketch.com/s/ac25729e-df9b-4e74-a09c-badb62787d24/a/ZVvo0ja#Inspect))
2. If ALL systems are Immutable, go to details with the Immutable tab in focus.
3. If there are NO immutable systems on the account, [Hide the tabs.](https://www.sketch.com/s/ac25729e-df9b-4e74-a09c-badb62787d24/a/kKdnAeg#Inspect)
4. Add tabs to the table if at least one Immutable system exists.
5. If there are no conventional systems but the account has immutable systems, keep the conventional tab with an empty state.
6. These columns exist: Name, Group, Tags, OS, Last Seen, First impacted
7. The system link goes to  “Inventory > System Details > Advisor tab 
8. Table can be filtered by Name, Tags, OS

# How to test the PR

Please include steps to test your PR.

# Before the change

# After the change

# Dependent work link

# Checklist:

- [ ] The commit message has the Jira ticket linked
- [ ] PR has a short description
- [ ] Screenshots before and after the change are added
- [ ] Tests for the changes have been added
- [ ] README.md is updated if necessary
- [ ] Needs additional dependent work

### CODE REVIEW

**Comment by adonispuente** [comment]
I've found a routing issue where if a click a pathway via the panel cards on the top, i can navigate to a page with your changes , https://stage.foo.redhat.com:1337/preview/insights/advisor/recommendations/pathways/systems/upgrade-kernel#SIDs=&tags= 
However If I select the pathways tab, and click the same pathway Im navigated to this page without your changes 
https://stage.foo.redhat.com:1337/preview/insights/advisor/recommendations/pathways/upgrade-kernel

**Comment by mkholjuraev** [comment]
@RedHatInsights/team-interact  The routing issues found by @adonispuente have been fixed. PTAL

File: src/SmartComponents/HybridInventoryTabs/ConventionalSystems/PathwaySystems.js [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+import { useDispatch } from 'react-redux';
+import Inventory from '../../../PresentationalComponents/Inventory/Inventory';
+
+const PathwaySystems = ({ pathway, selectedTags, workloads, SID }) => {
+  const dispatch = useDispatch();
+
+  useEffect(() => {
+    return () => {
+      dispatch({
+        type: 'CLEAR_INVENTORY_STORE',
```
Comment (bastilian) [inline src/SmartComponents/HybridInventoryTabs/ConventionalSystems/PathwaySystems.js:12]: @mkholjuraev What was the prop of the InventoryTable that made this redundant?

Comment (mkholjuraev) [reply src/SmartComponents/HybridInventoryTabs/ConventionalSystems/PathwaySystems.js]: It was `initialLoading`, removed this redux clear action. I have verified that the prop [exists](https://github.com/RedHatInsights/insights-advisor-frontend/blob/master/src/PresentationalComponents/Inventory/Inventory.js#L395) in the Inventory component. PTAL

File: src/SmartComponents/HybridInventoryTabs/HybridInventoryTabs.js [hunk 2]
```diff
@@ -37,15 +43,19 @@ const HybridInventory = ({
...
+            <RecommendationSystems {...tabProps} />
+          ) : (
+            <PathwaySystems {...tabProps} />
+          )}
         </Suspense>
       }
       ImmutableDevicesTab={
         <Suspense fallback={Fragment}>
           <ImmutableDevices {...tabProps} />
         </Suspense>
```
[the remark this answers is not in the document]
Comment (mkholjuraev) [reply src/SmartComponents/HybridInventoryTabs/HybridInventoryTabs.js:56]: @gkarat thank you for the review.
1. The requirement is a bit confusing at first glance, but what you have seen is expected behavior. The req says 'If there are NO immutable systems on **the account**', so we need to look at an account level, not the pathway level. As insights-qa account has immutable systems, the tabs should not be hidden.
2. Good catch, will fix it ASAP

File: src/SmartComponents/Recs/helpers.js [hunk 3]
```diff
@@ -73,33 +73,45 @@ export const bulkHostActions = async ({
...
+
 export const edgeSystemsCheck = async (
   ruleId,
   setSystemsCount,
   setEdgeSystemsCount,
   setConventionalSystemsCount,
-  setCountsLoading
+  setCountsLoading,
+  pathway
 ) => {
```
[the remark this answers is not in the document]
Comment (mkholjuraev) [reply src/SmartComponents/Recs/helpers.js:97]: Hmm, nice idea. I would prefer to do it in a separate PR not to bloat this PR and to be focused only ticket requirements

[the remark this answers is not in the document]
Comment (gkarat) [reply src/SmartComponents/Recs/helpers.js:97]: Makes sense, thank you

**Comment by adonispuente** [comment]
There are times when I check on a pathway with immutable systems that the additional tabs do not show up, as seen on this screen shot of the Sysctl Parameter Tuning pathway (that includes immutable systems)

Screenshot 2023-11-30 at 12 05 57 PM

**Comment by mkholjuraev** [comment]
> Also, for some reason when sort by last seen the arrow pops up twice, though im unsure how related this is to your PR Screenshot 2023-11-30 at 12 07 52 PM

This issue happens on a new number of apps and also in advisor. I suspect it needs to be fixed in the inventory.

**Comment by adonispuente** [comment]
Everything else seems to have been resolved. 
On the convetional tab there are 4 options : Names, Tags,OS, and Last Seen, im assuming this is what is referred too in 
8.Table can be filtered by Name, Tags, OS
But this table can only be filtered by Name and Last Seen
Screenshot 2023-11-30 at 1 27 23 PM

The Immutable tab has 8 columns as opposed to the 7 you have in the PR description, with the additional **Image** column. Is this intentional? 
Also the immutable age is only filterable by Name and Last Seen
Screenshot 2023-11-30 at 1 27 11 PM

**Comment by mkholjuraev** [comment]
@adonispuente thank you for having look.
1. I see all the 3 filters on my end and they are working
image
2. The columns that should be on the UI are: Name, Group, tags, OS, last seen, image status.

**Review by adonispuente — APPROVED** [review]
Sorry about to confusing, I had mixed up sorting vs filtering. LGTM! Great job

**Pull request merged**